### ISSUE TITLE
Downloading the Splashtop Streamer

### ISSUE THREAD

**Issue by precursorca** [homebysix/recipe-robot#96]
Trying to get the Splashtop Streamer using their URL: http://www.splashtop.com/downloadstart?platform=mac

This produces a .dmg file when one uses the URL in a browser.

Recipe Robot App says: 
"Opening downloaded file
"Warning: I've investigated pretty thoroughly, and I'm still not sure what the download format is. This could cause problems later.
"[ERROR] I wasn't able to determine the name of this app, so I can't make any recipe's.

**Comment by precursorca** [comment]
I have some better log info now. Seems like it doesn't like the App in the App folder. Nor the installer in the DMG.

I used a URL from my login account and it gave more info. (I have removed personal info in this string and replaced it with "blahblahblah")

Processing https://blahblahblah.cloudfront.net/csrs/Splashtop_Streamer_MAC_v3.0.6.3.dmg ...
Input path looks like a download URL.
    Download URL is: https://blahblahblah.cloudfront.net/csrs/Splashtop_Streamer_MAC_v3.0.6.3.dmg
[WARNING] Careful, this might be a version-specific URL. Better to give me a "-latest" URL or a Sparkle feed.
Downloading file for further inspection...
    Downloaded to /Users/blahblahblah/Library/Caches/Recipe Robot/2016-04-21_11-00-52_409544/Splashtop_Streamer_MAC_v3.0.6.3.dmg
Determining download format...
    File extension is dmg
Opening downloaded file...
    Successfully mounted disk image
Checking whether package is signed...
    Package is signed
Getting developer from pkg signature...
    Developer is: Splashtop Inc.
Getting package signature authority names...
    3 authority names recorded
Expanding package to look for clues...
    Package expanded to: /Users/blahblahblah/Library/Caches/Recipe Robot/2016-04-21_11-00-52_409544/expanded
Getting information from PackageInfo file...
    Bundle identifier: com.splashtop.splashtopStreamer.com.splashtop.streamer-daemon.pkg
    Install location: /Library/LaunchDaemons
    Install filename: LaunchDaemons
Extracting the package payload to see if we can find an app...
    Did not find an app in the package payload
Getting information from PackageInfo file...
    Install location: /Library/LaunchAgents
    Install filename: LaunchAgents
Extracting the package payload to see if we can find an app...
    Did not find an app in the package payload
Getting information from PackageInfo file...
    Install location: /Library/LaunchAgents
    Install filename: LaunchAgents
Extracting the package payload to see if we can find an app...
    Did not find an app in the package payload
Getting information from PackageInfo file...
    Install location: /Library/LaunchDaemons
    Install filename: LaunchDaemons
Extracting the package payload to see if we can find an app...
    Did not find an app in the package payload
Getting information from PackageInfo file...
    Install location: /Applications
    Install filename: Applications
Extracting the package payload to see if we can find an app...
    Did not find an app in the package payload
[ERROR] I wasn't able to determine the name of this app, so I can't make any recipes.
[ERROR]

**Comment by homebysix** [comment]
Hi @precursorca,

Thanks for reporting this. Let's talk programmatically about Recipe Robot might process a dmg like this one:

![screen shot 2016-04-21 at 8 02 49 am](https://cloud.githubusercontent.com/assets/7801391/14718294/b4f5451c-07aa-11e6-9bb0-59d375b834ba.png)

99% of the time, when you see an app inside a dmg, the correct procedure is to copy the app to your Applications folder. However, that wouldn't be the right thing to do here. Instead, we should install the pkg.

But we can't just blindly install pkg files inside dmgs either. Take this case, for example:

![screen shot 2016-04-21 at 8 09 29 am](https://cloud.githubusercontent.com/assets/7801391/14718896/15e8baea-07ae-11e6-8832-976e4413a057.png)

Which pkg do we install? We know because it's called "Install" but how do we translate that into reliable code?

Bottom line: Recipe Robot is meant to work well for basic Mac apps, which includes those that are distributed as an app wrapped in a dmg or zip. It may not work for more complex apps like Splashtop Streamer, and that's OK with me.
